athena/etc/xdm/console/config_console.1
.\" $Id: config_console.1,v 1.6 1999-01-22 23:16:08 ghudson Exp $
.TH CONFIG_CONSOLE 1 "16 November 1990"
.ds ]W MIT Project Athena
.SH NAME
config_console, show_console, hide_console \- reconfigure X console window
.SH SYNOPSIS
.nf
config_console [-hide] [-show] [-config]
hide_console [-hide] [-show] [-config]
show_console [-hide] [-show] [-config]
.fi
.SH DESCRIPTION
This program allows the user to manipulate the console window.  If
invoked as \fBconfig_console\fR or given the \fB-config\fR argument,
it will reconfigure the console.  This causes the console to re-read
its resources, re-grab the logical console for the workstation, and
hide itself.  To change the default behavior of the console, load your
console resources into the X server with ``xrdb -merge
\fIfilename\fR'', then invoke \fIconfig_console\fR.  See console(8)
for a list of resources that are honored.

If the program is called \fBhide_console\fR or it is invoked with the
\fB-hide\fR argument, instead of reconfiguring the console it will
make the window disappear until the next time it has something to
display.

If the program is called \fBshow_console\fR or it is invoked with the
\fB-show\fR argument, instead of reconfiguring the console it will
make the window reappear on the screen.

.SH FILES
/var/athena/console.pid

.SH "SEE ALSO"
console(8), dm(8), xrdb(1)

.SH AUTHOR
Mark Rosenstein, MIT Project Athena
.br
Copyright (c) 1990 Massachusetts Institute of Technology




doc/news
This file is intended to keep track of user-visible changes to the
release, for the construction of the Athena release notes.

Changes from release 8.0 to release 8.1:

	* Athena version numbers are now dotted triplets, e.g. 8.1.0
	  or 8.1.12.  Paths which include all or part of the Athena
	  version number now take the form "pathname-version",
	  e.g. "srvd-8.1" or "dash.menus-8.1"

	* In release 8.0, running "xmkmf" with the default path would
	  generate a Makefile which looked for its include files and
	  libraries in /mit/x11, which could be one of several
	  lockers.  Now it is preferrable to use the vendor X
	  libraries, so this bit of glue is going away.  There will be
	  no /usr/athena/bin/xmkmf any more, nor any symlinks in
	  /usr/athena pointing to the x11 locker.

	  Users can still do something like "add -f x11r5" and then
	  "xmkmf", and things should work, if the x11r5 locker is
	  built for their platform.

	* Athena bitmaps have moved to /usr/athena/include/X11/bitmaps.
	  Previously, they were in /usr/athena/lib/X11/bitmaps before
	  because of the /mit/x11 glue.

	* The symlink /bin/csh -> /bin/athena/tcsh has gone away.  The
	  update to 8.0 will fix /bin/csh references in /etc/passwd
	  and /etc/passwd.local to refer to /bin/athena/tcsh instead.

	* emacs18 has gone away.

	* AFS includes and libraries are installed in
	  /usr/afsws/include and /usr/afsws/lib.

	* wcl includes and libraries are installed in /usr/athena.

	* The basic krb5 headers and libraries are installed in
	  /usr/athena.

	* emacs has been upgraded from 19.30 to 19.34.  There should
	  be no significant user-visible changes.

	* RCS has been upgraded from version 4 to version 5.7.

	* x3270 has been upgraded from version 1.2 to version 3.1.1.3.

	* sysinfo is now available under Irix.

	* /bin/athena/tcsh now has Hesiod support enabled, which means
	  that "~lockername" will work in a pathname in most cases.
	  (It won't work for users with local passwd information who
	  don't have their lockers attached.)

	* For improved ease of debugging, many binaries are installed
	  with symbol tables, and compiled without optimization.
	  Therefore, the contents of /bin/athena and /usr/athena/bin
	  may be somewhat larger, and the binaries may consume more
	  CPU cycles, although it's unlikely that anyone will notice
	  the speed difference.


athena/etc/xdm/console/config_console.c
/* $Id: config_console.c,v 1.7 1999-01-22 23:16:09 ghudson Exp $
 *
 * Copyright (c) 1990 by the Massachusetts Institute of Technology
 * For copying and distribution information, please see the file
 * <mit-copyright.h>.
 *
 * This program allows a user to manipulate the console window.
 * This is done by sending signals to it.
 */

#include <stdio.h>
#include <signal.h>
#include <sys/types.h>
#include <sys/file.h>
#include <sys/stat.h>
#include <string.h>

char *pidfile = "/var/athena/console.pid";


main(argc, argv)
int argc;
char **argv;
{
    char *name, buf[256];
    FILE *f;
    struct stat before, after;
    int pid, i = 0;
    int sig = SIGHUP;

    name = strrchr(argv[0], '/');
    if (name == NULL)
      name = argv[0];
    else
      name++;

    if (!strcmp(name, "show_console"))
      sig = SIGUSR1;
    else if (!strcmp(name, "hide_console"))
      sig = SIGUSR2;	

    if (argc == 2 && !strcmp(argv[1], "-hide"))
      sig = SIGUSR2;
    else if (argc == 2 && !strcmp(argv[1], "-show"))
      sig = SIGUSR1;
    else if (argc == 2 && !strcmp(argv[1], "-config"))
      sig = SIGHUP;
    else if (argc != 1) {
	fprintf(stderr, "Usage: %s [-config | -show | -hide]\n", name);
	exit(2);
    }

    if (stat(pidfile, &before)) {
	sprintf(buf, "%s: unable to find console", name);
	perror(buf);
	fprintf(stderr, "\tlooking for file %s\n", pidfile);
	exit(1);
    }

    f = fopen(pidfile, "r");
    if (f == NULL) {
	sprintf(buf, "%s: unable to find console", name);
	perror(buf);
	fprintf(stderr, "\tlooking for file %s\n", pidfile);
	exit(1);
    }
    fgets(buf, sizeof(buf), f);
    fclose(f);

    pid = atoi(buf);
    if (pid < 2) {
	fprintf(stderr, "%s: unable to find proper console\n", name);
	fprintf(stderr, "\tconsole cannot have PID %d\n", pid);
	exit(1);
    }

    if (kill(pid, sig) < 0) {
	sprintf(buf, "%s: unable to signal console", name);
	perror(buf);
	exit(1);
    }

    if (sig != SIGHUP)
      exit(0);

    do {
	sleep(1);
	if (stat(pidfile, &after))
	  break;
    } while (before.st_mtime == after.st_mtime && i++ < 10);
    exit(0);
}
